pkg/controller/replication_controller.go

```diff
@@ -204,6 +204,10 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 // When a pod is created, enqueue the controller that manages it and update it's expectations.
 func (rm *ReplicationManager) addPod(obj interface{}) {
 	pod := obj.(*api.Pod)
+	if pod.DeletionTimestamp != nil {
```

Comment:
@bprashanth explained in person that the behavior is indeed different for actions initiated by the ReplicationManager, such as add and delete, vs. observing status changes, such as reports that pods have failed. So, I think we just need a comment explaining why this check is needed, at least for updatePod, so that the ReplicationManager doesn't just sleep until it has observed its own delete action.

Whether this is needed in addPod is questionable, but I don't object so long as it doesn't cause problems.